src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs

```diff
@@ -679,7 +679,9 @@ private static bool ElementNameIs(XElement element, string name)
             // check to see if we're referencing a symbol defined in source.
             static bool isSymbolDefinedInSource(Location l) => l.IsInSource;
             return symbols.WhereAsArray((s, arg) =>
-                (s.Locations.Any(isSymbolDefinedInSource) || !s.HasUnsupportedMetadata) &&
+                // Check if symbol is namespace (which is always visible) first to avoid realizing all locations
+                // of each namespace symbol, which might end up allocating in LOH
+                (s.IsNamespace() || s.Locations.Any(isSymbolDefinedInSource) || !s.HasUnsupportedMetadata) &&
```

Comment:
⚠️ Namespaces which are imported from metadata but do not contain any accessible symbols are not considered visible today (e.g. the `CodeGen` in `Microsoft.CodeAnalysis.CodeGen` is defined in Microsoft.CodeAnalysis.dll, but will not appear in completion within Microsoft.CodeAnalysis.Features.dll). Are we sure this code change will not impact this behavior?

Reply:
It doesn't look like this change would have impacted this behavior though, given that ` `s.Locations.Any(isSymbolDefinedInSource) || !s.HasUnsupportedMetadata` would be true for your example? I will validate and figure out where exactly that filtering is handled. Thanks for bringing this up!

Reply:
Added a test